Avatar shapes

ttseng opened the pull request:
https://avatar-shapes.glitch.me

Restyle user avatars & team avatars on profile and team pages to help distinguish between entities.

More info here: https://www.notion.so/glitch/Distinguish-between-user-avatar-vs-team-avatar-c475ed420dd444cba0b1ecadcca4e0a7

![profile page](https://user-images.githubusercontent.com/519336/48443914-90fa4e80-e760-11e8-918b-877b63b1fcab.png)

![team page](https://user-images.githubusercontent.com/519336/48443928-99528980-e760-11e8-92c3-338a4b6e1ed8.png)

judeallred commented:
spotted a couple of style quirks on the teams page:
- alignment of @ sign
- styles of the 'invited' display

![image](https://user-images.githubusercontent.com/12502380/48445094-b472c880-e763-11e8-9df2-af80ab4af9d7.png)

judeallred commented:
the user icon in the corner should probably be a circle too, yeah?
![image](https://user-images.githubusercontent.com/12502380/48445576-e0db1480-e764-11e8-913b-2535270b7b34.png)

ttseng commented:
yup, I just realized the user avatar in the header should also be round and pushed an update.
re: the alignment of @ sign, looks like that's an issue on production.  I'll fix it and look into the invited pop-over styling.

judeallred commented:
thanks

judeallred commented:
also these -- 
![image](https://user-images.githubusercontent.com/12502380/48446445-416b5100-e767-11e8-99eb-4c22b9670677.png)

judeallred commented:
also noticed that some of the inputs are a little smushed in avatar-shapes  (maybe because of the fix for the @ sign above :-p  )
![image](https://user-images.githubusercontent.com/12502380/48446528-7aa3c100-e767-11e8-995a-861e7ce75e7d.png)

ttseng commented:
looks like the inline-styling for teams is incorrect on production:

![image](https://user-images.githubusercontent.com/519336/48446598-acb52300-e767-11e8-85d2-4fb7b87c685a.png)

I think it might be best to move that fix into a separate PR.

ttseng commented:
A few more updates:

* Update `project-result-item` to include new user styling
<img width="404" alt="screen shot 2018-11-13 at 5 33 53 pm" src="https://user-images.githubusercontent.com/519336/48448183-323ad200-e76c-11e8-98b8-855a138a4cf5.png">

* Updated `add-team-user` to include new user styling and condense layout 
<img width="333" alt="screen shot 2018-11-13 at 5 46 23 pm" src="https://user-images.githubusercontent.com/519336/48448244-60b8ad00-e76c-11e8-8ce3-b9e4ced8b7bb.png">

ttseng commented:
also needs to be fixed in the embed, but I'm not sure where that code lives
![image](https://user-images.githubusercontent.com/519336/48493621-2695ec80-e7fa-11e8-8dcb-40701aabde9b.png)

ttseng commented:
updated styling on user profile page to remove border separating user info with team associations

![image](https://user-images.githubusercontent.com/519336/48568522-96c17280-e8cd-11e8-96fc-20c01249ad68.png)

ttseng commented:
Updated search results
![image](https://user-images.githubusercontent.com/519336/48568982-adb49480-e8ce-11e8-8cc4-ba9b331cc2f0.png)

ttseng commented:
fixed the glitch avatar, which seems to be its own category of avatar 
![image](https://user-images.githubusercontent.com/519336/48572922-d04bab00-e8d8-11e8-8026-a6baeec64bd4.png)

ttseng commented:
@GregWeil ohh I didn't realize it was the glitch team, I thought it was a Glitch user.  I'll fix it.

ttseng commented:
fixed wrapping of user avatars in team search results (thanks for the suggestion Greg!)
![image](https://user-images.githubusercontent.com/519336/48574252-b90ebc80-e8dc-11e8-8751-25f645598268.png)

also changed avatar shapes for projects owned by a team
![image](https://user-images.githubusercontent.com/519336/48574278-c9269c00-e8dc-11e8-9d2b-4bb7ceab6460.png)

GregWeil commented:
User page team hovers are off center
![image](https://user-images.githubusercontent.com/8932174/48585920-2f6ee700-e8fc-11e8-980c-da700024eb20.png)

pketh commented on styles/pages/user-and-team-page.styl:
```diff
@@ -20,4 +38,68 @@
+    margin-top: 3px
+    
+    
+  .beta
```
`.beta` doesn't seem to be referenced anywhere else in this pr. Is this for a seperate thing?

ttseng replied:
I think it has to do with the beta flag that @whimsicallyson added to team pages

whimsicallyson replied:
all my beta flag style work was in `team.styl`, so this isn't me.

ttseng replied:
@whimsicallyson I deleted the team.styl page because there was already team page styling in user-and-team-page.styl, and I thought having a separate team.styl file could be confusing.

whimsicallyson replied:
gotcha. so that's probably from combining the files :)

GregWeil commented on styles/styles.styl:
```diff
@@ -33,7 +33,6 @@
 @require 'includes/zine-items'
 @require 'pages/user-and-team-page'
 @require 'pages/project'
-@require 'pages/team'
```
This file should get deleted if it isn't referenced

ttseng replied:
I removed it and moved all its contents to an existing styl file `pages/user-and-team-page` to keep our files more organized

GregWeil replied:
`pages/team.styl` is still there though 😕

pketh commented:
the 'made with glitch' placeholder avatar isn't round
<img width="521" alt="screen shot 2018-11-15 at 5 27 28 pm" src="https://user-images.githubusercontent.com/877072/48586582-5201ff80-e8fe-11e8-8f60-10f4d0cb05a2.png">


the teams that I'm a part of are part of my identifying metadata just like name and description, they can be listed out just like other metadata without the awkwardness complexity of the lighter background and the fully left alignment. Most users won't belong to too many teams, or any teams, so the extra horizontal space of a separate bar isn't really needed.
<img width="505" alt="screen shot 2018-11-15 at 5 29 33 pm" src="https://user-images.githubusercontent.com/877072/48586579-50383c00-e8fe-11e8-8ea3-b0cdd2890a15.png">


this isn't a blocker, but with the circle avatars the 'upload avatar' button looks awk/ not as nice/integrated into the design anymore
<img width="207" alt="screen shot 2018-11-15 at 5 31 28 pm" src="https://user-images.githubusercontent.com/877072/48586576-4d3d4b80-e8fe-11e8-8dbb-df16339b6e9e.png">

ttseng commented:
Thanks for catching the tooltip misalignment @GregWeil - I fixed it:

<img width="495" alt="screen shot 2018-11-15 at 5 49 09 pm" src="https://user-images.githubusercontent.com/519336/48586941-6d213f00-e8ff-11e8-9f34-92c5efc82ca5.png">
<img width="527" alt="screen shot 2018-11-15 at 5 50 19 pm" src="https://user-images.githubusercontent.com/519336/48586944-6eeb0280-e8ff-11e8-93f4-f0ab3679813f.png">

ttseng commented:
@pketh I had originally changed the 'made with glitch' icon to round, but @GregWeil pointed out that clicking on it links to the Glitch team, so I think it should actually be rectangular with rounded corners like other team avatars.

Putting the team avatars below the user info, rather than below the user description, takes up less vertical space, especially for profiles like Jenn and yours, where you're part of lots of teams, so I prefer the existing design on avatar-shapes.glitch.me.

Here's a comparison side by side.

![image](https://user-images.githubusercontent.com/519336/48587079-da34d480-e8ff-11e8-9e9d-fa2f0a34632e.png)

ttseng commented:
re: the upload avatar button positioning, we could center it:
![image](https://user-images.githubusercontent.com/519336/48587221-5cbd9400-e900-11e8-8e06-4cbcf9e324bb.png)
 or even better IMO, we could just get rid of it trigger it when the user hovers over their existing avatar.  there might be accessibility issues with this, though - maybe @jennschiffer can comment

judeallred commented:
[for the upload avatar button]
My favorite options are centering it, or hiding it and exposing it (at the base of the image) on hover.  I agree it looks like a visual bug to keep it anchored to the left,  but I'm comfy with all of the fixes.

jennschiffer commented:
i personally like having the button there, but if we were to remove it or hide it, the avatar should itself be a button element or have a [button role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) if clicking it would trigger the upload. we should avoid hiding things behind hover states.

pketh commented:
ya I lean away from hiding it for both initial affordance reasons, and because hover only doesn't work on touch devices

pketh commented:
most users won't be like jenn and me or even be part of teams so I wouldn't optimize with our needs in mind. Of the 400+ teams currently created, most have only a single user. That'll go up but I wouldn't expect a user belonging to 4+ teams ever being hyper common, using github usage as an example.

ttseng commented:
Centered upload button underneath avatars
![image](https://user-images.githubusercontent.com/519336/48649429-91912000-e9c0-11e8-963b-0fe8802fc39d.png)
![image](https://user-images.githubusercontent.com/519336/48649431-95bd3d80-e9c0-11e8-882b-44f8f011ab6f.png)
